LibraryManagement/src/main/java/com/libraryManagement/entity/Room.java

```diff
@@ -1,5 +1,53 @@
 package com.libraryManagement.entity;
+import java.util.List;
 
-public class Room {
+import com.fasterxml.jackson.annotation.JsonManagedReference;
 
-}
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+@Entity
+public class Room {
+		@Id
+		@GeneratedValue(strategy = GenerationType.IDENTITY)
+		private int roomId;
+		
+		private String roomName;
+		
+		@OneToMany(mappedBy = "room" , cascade = CascadeType.ALL,orphanRemoval = true)
+		@JsonManagedReference
+		private List<Row> rows;
+		
+		
+		public int getRoomId() {
+			return roomId;
+		}
+		public void setRoomId(int roomId) {
+			this.roomId = roomId;
+		}
+		public String getRoomName() {
+			return roomName;
+		}
+		public void setRoomName(String roomName) {
+			this.roomName = roomName;
+		}
+		public List<Row> getRows() {
+			return rows;
+		}
+		public void setRows(List<Row> rows) {
+			this.rows = rows;
+		}
+		public Room(int roomId, String roomName, List<Row> rows) {
+			super();
+			this.roomId = roomId;
+			this.roomName = roomName;
+			this.rows = rows;
+		}
+		public Room() {
```

Comment:
why you added super ?

Reply:
Sometime we created object without field ,that's why we created one empty constructor in entity.